crates/simd-doc/src/tests/fixtures.rs

```diff
@@ -166,76 +167,91 @@ fn test_incomplete_row_parsing() {
             }
         }
 
-        debug_assert_eq!(parse_snap, transcode_snap);
+        assert_eq!(parse_snap, transcode_snap);
 
         snaps.push((String::from_utf8(real_input).unwrap(), parse_snap));
     }
 
     insta::assert_debug_snapshot!(snaps, @r###"
-      [
-          (
-              "st\": {\"this\": \"is\", \"a\": \"doc\"}}\n{\"this\": \"is a real doc\"}\n",
-              [
-                  (
-                      -1,
-                      String("expected value at line 1 column 1 @ 0..33"),
-                  ),
-                  (
-                      59,
-                      Object {
-                          "this": String("is a real doc"),
-                      },
-                  ),
-              ],
-          ),
-          (
-              "\"test\": {\"this\": \"is\", \"a\": \"doc\"}}\n{\"this\": \"is a real doc\"}\n",
-              [
-                  (
-                      -1,
-                      String("incomplete row @ 0..45"),
-                  ),
-                  (
-                      62,
-                      Object {
-                          "this": String("is a real doc"),
-                      },
-                  ),
-              ],
-          ),
-          (
-              "{\"test\": 5}, {\"test\": 6\"}]\n{\"this\": \"is a real doc\"}\n",
-              [
-                  (
-                      -1,
-                      String("incomplete row @ 0..33"),
-                  ),
-                  (
-                      53,
-                      Object {
-                          "this": String("is a real doc"),
-                      },
-                  ),
-              ],
-          ),
-          (
-              "{\"real\": \"object\"}\n{\"this\": \"is a real doc\"}\n",
-              [
-                  (
-                      19,
-                      Object {
-                          "real": String("object"),
-                      },
-                  ),
-                  (
-                      45,
-                      Object {
-                          "this": String("is a real doc"),
-                      },
-                  ),
-              ],
-          ),
-      ]
+    [
+        (
+            "st\": {\"this\": \"is\", \"a\": \"doc\"}}\n{\"this\": \"is a real doc\"}\n",
+            [
+                (
+                    -1,
+                    String("expected value at line 1 column 1 @ 0..33"),
+                ),
+                (
+                    59,
+                    Object {
+                        "this": String("is a real doc"),
+                    },
+                ),
+            ],
+        ),
+        (
+            "\"test\": {\"this\": \"is\", \"a\": \"doc\"}}\n{\"this\": \"is a real doc\"}\n",
+            [
+                (
+                    -1,
+                    String("trailing characters at line 1 column 7 @ 0..36"),
```

Comment:
Where does this message come from? It doesn't match the new errors you added to the parser: `missing trailing newline`

Reply:
Yep. The Parser attempts to use simd-json first, which almost always works. If simd-json fails, though, we throw a c++ exception which gets turned into a Result by the `cxx` crate.

We handle this by using a fallback path which does the same job but using serde_json, which produces better errors and is able to handle documents of larger sizes (simd-json is constrained to 1MB documents).

So tl;dr these errors are produced by the serde_json fallback after the "missing excepted newline" exception is thrown / caught / handled.